src/extensions/extension-discovery.ts

```diff
@@ -266,18 +266,30 @@ export class ExtensionDiscovery {
 
     logger.info(`${logModule} loading extensions from ${extensionInstaller.extensionPackagesRoot}`);
 
-    if (fs.existsSync(path.join(extensionInstaller.extensionPackagesRoot, "package-lock.json"))) {
+    if (await fs.pathExists(path.join(extensionInstaller.extensionPackagesRoot, "package-lock.json"))) {
       await fs.remove(path.join(extensionInstaller.extensionPackagesRoot, "package-lock.json"));
     }
 
     try {
+      // Verify write access to static/extensions, which is needed for symlinking
       await fs.access(this.inTreeFolderPath, fs.constants.W_OK);
```

Comment:
The docs for this specifically call out using `access` for cases like this as race conditions.

Reply:
I think our usage is a bit far from the case the docs explain. In our case if the app would _not_ have write access, but then 1ms later it magically would, we could _still_ use the alternative code path, which works but I suppose is slower. If you mean the case that we have write access now but then lose it, you're correct.

However, I think neither of these cases of the folder rights changing during runtime are happening in practice.

I guess the theoretical way to handle this is to run npm install, catch the error of "symlinking failed because of no write access" and then do the second path. No idea how possible that's in practice though.

Reply:
The theoretical way would be the way I would want to do it, but yes. I don't know how to catch that case either.